AppCenterDataStorage/AppCenterDataStorage/Internal/LocalStore/MSDBDocumentStore.m

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import <sqlite3.h>
+
+#import "AppCenter+Internal.h"
+#import "MSAppCenterInternal.h"
+#import "MSConstants+Internal.h"
+#import "MSDBDocumentStorePrivate.h"
+#import "MSDBStoragePrivate.h"
+#import "MSDataStore.h"
+#import "MSDataStoreInternal.h"
+#import "MSDocumentWrapper.h"
+#import "MSUtility+StringFormatting.h"
+#import "MSWriteOptions.h"
+
+static const NSUInteger kMSSchemaVersion = 1;
+
+@implementation MSDBDocumentStore
+
+#pragma mark - Initialization
+
+- (instancetype)init {
+
+  /*
+   * DO NOT modify schema without a migration plan and bumping database version.
+   */
+  MSDBSchema *schema = @{kMSAppDocumentTableName : [MSDBDocumentStore tableSchema]};
+  if ((self = [super init])) {
+    self.dbStorage = [[MSDBStorage alloc] initWithSchema:schema version:kMSSchemaVersion filename:kMSDBDocumentFileName];
+    NSDictionary *columnIndexes = [MSDBStorage columnsIndexes:schema];
+    _idColumnIndex = ((NSNumber *)columnIndexes[kMSAppDocumentTableName][kMSIdColumnName]).unsignedIntegerValue;
+    _partitionColumnIndex = ((NSNumber *)columnIndexes[kMSAppDocumentTableName][kMSPartitionColumnName]).unsignedIntegerValue;
+    _documentIdColumnIndex = ((NSNumber *)columnIndexes[kMSAppDocumentTableName][kMSDocumentIdColumnName]).unsignedIntegerValue;
+    _documentColumnIndex = ((NSNumber *)columnIndexes[kMSAppDocumentTableName][kMSDocumentColumnName]).unsignedIntegerValue;
+    _eTagColumnIndex = ((NSNumber *)columnIndexes[kMSAppDocumentTableName][kMSETagColumnName]).unsignedIntegerValue;
+    _expirationTimeColumnIndex = ((NSNumber *)columnIndexes[kMSAppDocumentTableName][kMSExpirationTimeColumnName]).unsignedIntegerValue;
+    _downloadTimeColumnIndex = ((NSNumber *)columnIndexes[kMSAppDocumentTableName][kMSDownloadTimeColumnName]).unsignedIntegerValue;
+    _operationTimeColumnIndex = ((NSNumber *)columnIndexes[kMSAppDocumentTableName][kMSOperationTimeColumnName]).unsignedIntegerValue;
+    _pendingOperationColumnIndex = ((NSNumber *)columnIndexes[kMSAppDocumentTableName][kMSPendingDownloadColumnName]).unsignedIntegerValue;
+  }
+  return self;
+}
+
+#pragma mark - Table Management
+
+- (bool)createWithPartition:(nonnull NSString *)__unused partition
```

Comment:
Since `NS_ASSUME_NONNULL_BEGIN` is declared in the header file, you don't `nonnull` in the parameters.